src/Domain/SurfaceElement.hpp

```diff
@@ -0,0 +1,61 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/Gsl.hpp"
+
+/*!
+ * \brief Compute the Euclidean surface element from the inverse jacobian.
+ *
+ * \details The Euclidean surface element of a surface \f$\Sigma\f$ with
+ * constant logical coordinate \f$\xi^i\f$ is given by: \f{equation} J^\Sigma =
+ * J \sqrt{\delta^{jk} (J^{-1})^i_j (J^{-1})^i_k} \f} where \f$J^i_j = \partial
+ * x^i / \xi^j\f$ is the volume Jacobian with determinant \f$J\f$ and inverse
+ * \f$(J^{-1})^i_j = \partial \xi^i / \partial x^j\f$. The determinant of the
+ * inverse Jacobian can be passed as an argument as an overload, otherwise it
+ * will be calculated.
+ *
+ * \note The curved space surface element as well as time dependent maps are not
+ * implemented yet but can be added on demand.
```

Comment:
No, I don't need those for now. The worldtube currently  represents the internal solution inside the worldtube in the grid frame and uses a euclidean coordinates.

I might need to add them later for non-circular orbits. The overloads should probably be the same ones as for `unnormalized_normal_vector`.

Reply:
Would you mind adding an overload that takes a spatial metric? We'll need that sooner or later. Do you currently get by with the Euclidean surface integral because you're working with a perturbative solution in the worldtube? Would it be wrong to use the spatial metric in the integral?

Reply:
Yes, the local solution is in Cartesian coordinates. I can add the proper curved space integral in a follow-up PR. Do you happen to know about a nice test  for that  like an analytical solution for a scalar quantity integrated over a sphere centered on a Schwarzschild/Kerr black hole?

Reply:
Yea the irreducible mass of a Kerr horizon is a good quantity to compute [Eq. (1.70) in Baumgarte/Shapiro]. It's also computed for a Strahlkorper here: https://github.com/sxs-collaboration/spectre/blob/develop/tests/Unit/ApparentHorizons/Test_StrahlkorperGr.cpp#L928